\Inflector::classify() assume parameter string is in plural, assigning $singular might cause improper result

crynobone opened the pull request:
\Inflector::classify() assume parameter string is in plural, assigning $singular might cause improper result:

e.g: $singular = 'campus';
It will return model as 'Model_Campu'

crynobone committed:
\Inflector::classify() assume parameter string is in plural, assigning $singular might cause improper result:

e.g: $singular = 'campus';
It will return model as 'Model_Campu'

jschreuder commented:
This isn't the right fix, there's already a fix for the campus/status type words in the dev branch but you are right - it would be good if the method would be aware of singular/plural/uncountable words and only converts if it's the wrong type.

crynobone commented:
I not sure whether that is related, yes the update with Inflector::pluralize() for status/campus is now valid, however in this case it's different since you trying to singularize a singular word, this the flow of the code
1. Get the name of desired model (in singular)
2. Generate the Model name using \Inflector::classify() <-- in this method the string will be singularize again.

crynobone committed:
'php oil generate config name' now will allow us to copy config from core/config/name.php to app/config/name.pho

crynobone committed:
Add two (2) thing:
- Combine all config from core and app
- Allow to add own config using parameter

crynobone committed:
Fix minor bug and add -overwrite option to force update config file

crynobone committed:
Standardize output from Oil
